openwisp_users/tests/test_admin.py

```diff
@@ -511,6 +511,119 @@ def test_change_staff_with_group(self):
         self.assertEqual(user.groups.count(), 1)
         self.assertEqual(user.groups.get(name='Administrator').pk, group.pk)
 
+    def test_staff_can_edit_staff(self):
+        user1 = self._create_user(
+            username="user1", email="email1@mail.com", is_staff=True
+        )
+        user2 = self._create_user(
+            username="user2", email="email2@mail.com", is_staff=True
+        )
+        org = self._get_org()
+        self._create_org_user(user=user1, organization=org, is_admin=True)
+        self._create_org_user(user=user2, organization=org, is_admin=True)
+        group = Group.objects.get(name='Administrator')
+        user1.groups.add(group)
+        user2.groups.add(group)
+        self.client.force_login(user1)
+        params = user2.__dict__
+        params['username'] = 'newuser1'
+        params['groups'] = str(group.pk)
+        params.pop('phone_number')
+        params.pop('_password')
+        params.pop('last_login')
+        params.update(self.add_user_inline_params)
+        params.update(self._additional_params_add())
+        params.update(self._get_edit_form_inline_params(user2, org))
+        path = reverse(f'admin:{self.app_label}_user_change', args=[user2.pk])
+        r = self.client.post(path, params, follow=True)
+        self.assertEqual(r.status_code, 200)
+        self.assertNotContains(r, 'error')
+        user2.refresh_from_db()
+        self.assertEqual(user2.username, 'newuser1')
+
+    def test_staff_can_edit_org_owner(self):
+        user1 = self._create_user(
+            username="user1", email="email1@mail.com", is_staff=True
+        )
+        user2 = self._create_user(
+            username="user2", email="email2@mail.com", is_staff=True
+        )
+        org = self._get_org()
+        org_user2 = self._create_org_user(user=user2, organization=org, is_admin=True)
+        org_owner = self._create_org_owner(
+            organization=org, organization_user=org_user2
+        )
+        self._create_org_user(user=user1, organization=org, is_admin=True)
+        group = Group.objects.filter(name='Administrator')
+        user1.groups.set(group)
+        user2.groups.set(group)
+        self.client.force_login(user1)
+        path = reverse(f'admin:{self.app_label}_user_change', args=[user2.pk])
+        r = self.client.get(path)
+        self.assertEqual(r.status_code, 200)
+        self.assertContains(r, f'class="readonly">{user2.username}')
```

Comment:
please also check for the presence of the warning we just added (`You do not have permission to edit or delete this user because they are owner of an organization`).